### PR TITLE
[KernelGen][Nvidia] Add batch_norm_update_stats operator with Triton kernel

### PR DESCRIPTION
**Pull request by LoserCheems**
## Summary
Adds a Triton kernel for `batch_norm_update_stats`. Computes per-channel mean and variance from input tensor using Welford's online algorithm for numerical stability, and optionally updates running statistics (running_mean with biased variance momentum, running_var with unbiased variance momentum).

Returns `(mean, var_biased)` both float32 shape `[C]`.

## Testing
- Parametrized tests over `6 shapes × 3 dtypes × 2 running_stats modes` (36 cases)
- Validated against PyTorch CUDA reference
- Tested on: Nvidia H20

## Performance
Test command: `pytest benchmark/test_batch_norm_update_stats.py --level core` (NVIDIA H20)

**float16:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [4, 16, 64, 4] | 0.0098 | 0.0067 | 1.464 |
| [16, 16, 8, 48] | 0.0165 | 0.0113 | 1.460 |
| [16, 16, 8, 88] | 0.0230 | 0.0165 | 1.394 |
| [16, 16, 128] | 0.0107 | 0.0075 | 1.432 |
| [20, 6, 65536] | 1.035 | 1.095 | 0.945 |

**float32:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [4, 16, 64, 4] | 0.0097 | 0.0071 | 1.360 |
| [16, 16, 8, 48] | 0.0164 | 0.0123 | 1.342 |
| [16, 16, 8, 88] | 0.0229 | 0.0168 | 1.366 |
| [16, 16, 128] | 0.0109 | 0.0083 | 1.318 |
| [20, 6, 65536] | 1.032 | 1.221 | 0.845 |

**bfloat16:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [4, 16, 64, 4] | 0.0099 | 0.0067 | 1.467 |
| [16, 16, 8, 48] | 0.0164 | 0.0117 | 1.395 |
| [16, 16, 8, 88] | 0.0228 | 0.0163 | 1.399 |
| [16, 16, 128] | 0.0107 | 0.0080 | 1.341 |
| [20, 6, 65536] | 1.037 | 1.102 | 0.942 |

Note: The [20, 6, 65536] shape has only 6 channels, limiting GPU parallelism (only 6 thread blocks). This is an edge case where the small channel count dominates. For typical BN workloads (16+ channels), speedups are consistently 1.3-1.5x.

## Files Changed
- `src/flag_gems/ops/batch_norm_update_stats.py`: Triton kernel (Welford's algorithm)
- `tests/test_batch_norm_update_stats.py`: Accuracy test
- `benchmark/test_batch_norm_update_stats.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.5)

### CODE REVIEW

**Comment by bin913** [comment]
@LoserCheems Please fix failing checks

**Comment by LoserCheems** [comment]
@bin913 fixed failing check

File: tests/test_batch_norm_update_stats.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        running_var = None
+
+    ref_out = torch.batch_norm_update_stats(
+        input_t,
+        running_mean.clone() if running_mean is not None else None,
+        running_var.clone() if running_var is not None else None,
+        0.1,
+    )
+
+    res_out = torch.batch_norm_update_stats(
```
Comment (bin913) [inline tests/test_batch_norm_update_stats.py:58]: Please ensure that the flags operator is called during testing

Comment (LoserCheems) [reply tests/test_batch_norm_update_stats.py]: done
